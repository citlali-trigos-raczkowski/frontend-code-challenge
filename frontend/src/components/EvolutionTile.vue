<template>
  <router-link id="EvolutionTile" v-bind:to="pokemon.name"
    >visit squirtle

    <v-app id="inspire">
      <v-card max-width="200">
        <v-img v-bind:src="pokemon.image" height="auto"></v-img>
        <v-card-title>{{ pokemon.name }}</v-card-title>
        <v-card-actions>
          <v-btn color="orange lighten-2" text> Explore </v-btn>
        </v-card-actions>
      </v-card>
    </v-app>
  </router-link>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "EvolutionTile",
  props: {
    pokemon: {
      type: Object,
      default: () => ({}),
    },
  },
});
</script>
