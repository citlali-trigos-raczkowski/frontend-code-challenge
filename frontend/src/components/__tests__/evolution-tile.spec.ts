import { sampleEvolutionPokemon } from "../../../tests/mocks/mock-props";
import { shallowMount, createLocalVue } from "@vue/test-utils";
import EvolutionTile from "../evolution-tile.vue";
import VueRouter from "vue-router";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("components/evolution-tile.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(EvolutionTile, {
      propsData: { pokemon: sampleEvolutionPokemon },
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("button opens up detail view", async () => {
    const wrapper = shallowMount(EvolutionTile, {
      propsData: { pokemon: sampleEvolutionPokemon },
      localVue,
    });

    const aWrapper = wrapper.find("router-link-stub");
    expect(aWrapper.attributes().to).toBe(sampleEvolutionPokemon.name);
  });
});
