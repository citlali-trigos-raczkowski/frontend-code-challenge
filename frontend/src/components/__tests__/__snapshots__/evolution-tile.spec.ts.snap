// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/evolution-tile.vue renders correctly 1`] = `
<router-link-stub
  ariacurrentvalue="page"
  event="click"
  id="evolution-tile"
  tag="a"
  to="Ivysaur"
>
  <v-card-stub
    loaderheight="4"
    maxwidth="200"
    tag="div"
  >
    <v-img-stub
      height="200"
      options="[object Object]"
      position="center center"
      src="https://img.pokemondb.net/artwork/ivysaur.jpg"
      transition="fade-transition"
    />
     
    <div
      id="evolution-text"
    >
      <v-card-title-stub>
        Ivysaur
      </v-card-title-stub>
       
      <v-icon-stub
        color="orange lighten-2"
      >
        mdi-arrow-right
      </v-icon-stub>
    </div>
  </v-card-stub>
</router-link-stub>
`;
